test/de/schildbach/pte/live/TpgProviderLiveTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright the original author or authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.pte.live;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Date;
+
+import org.junit.Test;
+
+import de.schildbach.pte.TpgProvider;
+import de.schildbach.pte.dto.Location;
+import de.schildbach.pte.dto.LocationType;
+import de.schildbach.pte.dto.NearbyLocationsResult;
+import de.schildbach.pte.dto.Point;
+import de.schildbach.pte.dto.QueryDeparturesResult;
+import de.schildbach.pte.dto.QueryTripsResult;
+import de.schildbach.pte.dto.SuggestLocationsResult;
+
+/**
+ * @author Andreas Schildbach
+ */
+public class TpgProviderLiveTest extends AbstractProviderLiveTest {
+    public TpgProviderLiveTest() {
+        super(new TpgProvider(secretProperty("tpg.api_authorization")));
+    }
+
+    @Test
+    public void nearbyStationsByCoordinate() throws Exception {
+        final NearbyLocationsResult result = queryNearbyStations(Location.coord(46209460, 6167140)); // Parc et plage des Eaux-Vives
+        print(result);
+    }
+
+    @Test
+    public void queryDepartures() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8501238", false); // Genève-Paquis
+        print(result);
+    }
+
+    @Test
+    public void queryDeparturesSuburbanTrain() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8516155", false); // Lancy-Pont-Rouge
+        print(result);
+    }
+
+    @Test
+    public void queryDeparturesTram() throws Exception {
+        final QueryDeparturesResult result = queryDepartures("8587907", true); // Plainpalais
+        print(result);
+    }
```

Comment:
I'm not sure what this true/false is for, but with false, it throws because number of results != 1

Reply:
The JavaDoc tells:

```
@param equivs also query equivalent stations?
```

This is used by the public transport companies for stations that are so close to each other that they could be regarded as "one station". A typical example is different sections of a main station (East Hall, West Hall, Subway, etc.).

Reply:
ok Geneva is particular, they split a single station in several spot. I think this is the reason